mypy.ini

```diff
@@ -1,19 +1,28 @@
 [mypy]
 strict = True
 disallow_untyped_calls = False
-exclude = ["tests"]
```

Comment:
turns out, this is a regex field and was making mypy avoid running on most source files.

mypy was reporting to be running on 3 source files; after these adjustments, it is reporting to run on 70 source files, and pointed out all the additional issues fixed in this PR.